Panic when scanning block with missing files

**Describe the bug**

`tempo-cli` (version 1.5.0) produces a panic when trying to scan the block contents of an incomplete (?) block – "incomplete" meaning a compacted block only having a `meta.compacted.json` object (no `data`, no `index`, no `bloom-`, no `meta.json; MinIO/S3 storage).

**To Reproduce**
Steps to reproduce the behavior:
1. Start MinIO and configure it as storage for compacted blocks
2. Delete objects  `data`, `index`, `bloom-*`, and `meta.json` from a compacted block, only leaving `meta.compacted.json` in the block
3. List the block with the `--scan` option: `tempo-cli list block single-tenant <block-id> -c <config.yml> --scan`

**Expected behavior**

No panic, only an error

**Environment:**

- Version 1.5.0
- Monolithic, but only using `tempo-cli`
- S3 storage (MinIO)

 **Additional Context**

```bash
$ cat config.yml 
storage:
  trace:
    backend: s3
    s3:
      bucket: tempo
      endpoint: localhost:9000
      insecure: true
```
```bash
$ tempo-cli list block single-tenant e50b76cc-241e-451d-a33a-a517c6e18e07 -c config.yml --scan
ID            :  e50b76cc-241e-451d-a33a-a517c6e18e07
Version       :  v2
Total Objects :  170
Data Size     :  266 kB
Encoding      :  zstd
Level         :  2
Window        :  463758
Start         :  2022-11-27 06:38:29 +0000 UTC
End           :  2022-11-27 06:58:53 +0000 UTC
Duration      :  20m24s
Age           :  1616h32m58s
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14d0946]

goroutine 1 [running]:
github.com/alecthomas/kong.catch(0xc0009a5eb8)
	github.com/alecthomas/kong@v0.2.11/kong.go:366 +0xa5
panic({0x1649e80, 0x290f660})
	runtime/panic.go:838 +0x207
main.dumpBlock({0x1ced490, 0xc000705f70}, {0x1ce1798, 0xc0005a4600}, {0xc0008e0440, 0xd}, 0x0?, {0xc0002beff0, 0x24}, 0x1)
	github.com/grafana/tempo/cmd/tempo-cli/cmd-list-block.go:86 +0xa86
main.(*listBlockCmd).Run(0x29424d0, 0x1?)
	github.com/grafana/tempo/cmd/tempo-cli/cmd-list-block.go:50 +0x71
reflect.Value.call({0x16210c0?, 0x29424d0?, 0x18?}, {0x187a7e0, 0x4}, {0xc00090e2d0, 0x1, 0x0?})
	reflect/value.go:556 +0x845
reflect.Value.Call({0x16210c0?, 0x29424d0?, 0xa?}, {0xc00090e2d0, 0x1, 0x1})
	reflect/value.go:339 +0xbf
github.com/alecthomas/kong.callMethod({0x187a0d1, 0x3}, {0x175a8e0?, 0x29424d0?, 0x3?}, {0x16210c0?, 0x29424d0?, 0x0?}, 0x0?)
	github.com/alecthomas/kong@v0.2.11/callbacks.go:71 +0x4e7
github.com/alecthomas/kong.(*Context).RunNode(0xc00012c700, 0xc00088c0d0, {0xc000883f40, 0x1, 0x1})
	github.com/alecthomas/kong@v0.2.11/context.go:610 +0x5f7
github.com/alecthomas/kong.(*Context).Run(0x158af00?, {0xc000883f40?, 0xc000883f50?, 0x60?})
	github.com/alecthomas/kong@v0.2.11/context.go:627 +0xb8
main.main()
	github.com/grafana/tempo/cmd/tempo-cli/main.go:82 +0xfa
```

--- a/cmd/tempo-cli/cmd-list-block.go
+++ b/cmd/tempo-cli/cmd-list-block.go
@@ -81,8 +81,8 @@ func dumpBlock(r tempodb_backend.Reader, c tempodb_backend.Compactor, tenantID s
 	fmt.Println("Age           : ", fmt.Sprint(time.Since(unifiedMeta.EndTime).Round(time.Second)))
 
 	if scan {
-		if meta.Version != v2.VersionString {
-			return fmt.Errorf("cannot scan block contents. unsupported block version: %s", meta.Version)
+		if unifiedMeta.Version != v2.VersionString {
+			return fmt.Errorf("cannot scan block contents. unsupported block version: %s", unifiedMeta.Version)
 		}
 
 		fmt.Println("Scanning block contents.  Press CRTL+C to quit ...")
@@ -154,7 +154,7 @@ func dumpBlock(r tempodb_backend.Reader, c tempodb_backend.Compactor, tenantID s
 
 			copy(prevID, objID)
 
-			trace, err := model.MustNewObjectDecoder(meta.DataEncoding).PrepareForRead(obj)
+			trace, err := model.MustNewObjectDecoder(unifiedMeta.DataEncoding).PrepareForRead(obj)
 			if err != nil {
 				return err
 			}

